modules/core/client/app/init.js

```diff
@@ -36,11 +36,13 @@ angular.module(ApplicationConfiguration.applicationModuleName).run(function ($ro
         }
       }
     }
+
   });
 
   // Record previous state
   $rootScope.$on('$stateChangeSuccess', function (event, toState, toParams, fromState, fromParams) {
     storePreviousState(fromState, fromParams);
+    $rootScope.$state = $state;
```

Comment:
Why do we need this? I'm not saying it's wrong, just would like to have some context.